include/oneapi/dpl/pstl/hetero/dpcpp/sycl_defs.h

```diff
@@ -53,6 +53,7 @@
 #define _ONEDPL_SYCL2023_ATOMIC_REF_PRESENT (_ONEDPL_LIBSYCL_VERSION >= 50500)
 #define _ONEDPL_SYCL_SUB_GROUP_MASK_PRESENT (SYCL_EXT_ONEAPI_SUB_GROUP_MASK >= 1) && (_ONEDPL_LIBSYCL_VERSION >= 50700)
 #define _ONEDPL_SYCL_PLACEHOLDER_HOST_ACCESSOR_DEPRECATED (_ONEDPL_LIBSYCL_VERSION >= 60200)
+#define _ONEDPL_SYCL_DEVICE_COPYABLE_SPECIALIZATION_BROKEN (_ONEDPL_LIBSYCL_VERSION < 70100)
```

Comment:
Probably the most important remaining question. Is this macro for Intel specific SYCL implementation? If no, should we also check that if it's Intel compiler or not?

Reply:
Yes, this is specific to the intel compiler implementation of sycl.  However, it was my understanding that  `__LIBSYCL_MINOR_VERSION`, `__LIBSYCL_MINOR_VERSION` , `__LIBSYCL_PATCH_VERSION` were specific to the intel compiler's implementation of the sycl runtime.  

Are these macros populated by other sycl runtime implementations?

If not, then I think there does not need to be any check.  The `else` should be the version which follows the spec, and where we end up if `_ONEDPL_LIBSYCL_VERSION` is undefined.

Is your understanding different on this?

Reply:
Good.

Reply:
Ah, ok.  They are definitely not defined by the spec.  I am not aware of any other place they would be coordinated between implementors.  I think they are merely updated with intel implementation of sycl runtime within intel/llvm 

https://intel.github.io/llvm-docs/PreprocessorMacros.html
I think what we have here is correct.  There was an earlier discussion of this here https://github.com/oneapi-src/oneDPL/pull/1521#discussion_r1572752995

Reply:
Ive realized I was partially wrong.  If the version macros are not populated then `_ONEDPL_LIBSYCL_VERSION` is set to `0`.  This would put non-intel sycl runtime implementations in the wrong branch.  I've adjusted the macro to exclude `0`.  We could change it to explicitly check for the intel compiler, but this more directly checks for the presence of the sycl runtime library headers.

Reply:
Yeah, after investigation I agree. `!= 0` is a good addition. I grepped the code and there are two other places where we check `_ONEDPL_LIBSYCL_VERSION` less than something. I don't propose to fix it in this PR, though.